tests/constraints/tests.py

```diff
@@ -995,6 +995,42 @@ def test_validate_expression_str(self):
             exclude={"name"},
         )
 
+    def test_validate_nullable_textfield_with_isnull_true(self):
+        is_null_constraint = models.UniqueConstraint(
+            "price",
+            "discounted_price",
+            condition=models.Q(unit__isnull=True),
```

Comment:
Would it make sense to have a constraint that uses `Q() | Q()` to see if it propagates through properly?

Reply:
If we want to test `sql.where.WhereNode.as_sql` [specificity](https://github.com/django/django/blob/34ca1d3ab45f8f3f3d5bb867e4d13ca663deaa89/django/db/models/sql/where.py#L115-L187) I think we should do it directly and not through `Constraint` usage which is just one of the many feature that relies on this behaviour.

Reply:
> Would it make sense to have a constraint that uses `Q() | Q()` to see if it propagates through properly?

This should be already covered by:
https://github.com/django/django/blob/34ca1d3ab45f8f3f3d5bb867e4d13ca663deaa89/tests/constraints/models.py#L22-L26

Reply:
Indeed, thanks!